easybuild/framework/easyblock.py

```diff
@@ -2428,18 +2428,28 @@ def extensions_step(self, fetch=False, install=True):
         if self.skip:
             self.skip_extensions()
 
-        exts_cnt = len(self.ext_instances)
-        for idx, ext in enumerate(self.ext_instances):
+        # Create progress bar for extensions
+        if self.progress_bar:
+            ext_task_id = self.progress_bar.add_task("Extension", total=len(self.ext_instances), software="")
+            # Step size to advance progress bar for extensions
+            ext_step_size = 1.0 / 4.0
```

Comment:
I see why you break up the installation of a single extension in 4 (prepare, pre-run, run, post-run), but perhaps that's a bit too fine-grained, since the prepare/pre-run and post-run are usually very quick (or empty) steps...